### DISCUSSION TITLE
How to render two or more instances of reactflow in react/typescript

### DISCUSSION THREAD

**Discussion by saiGaneshAtmecs** [xyflow/xyflow#4157] [Q&A]
Hi Team, I'm using reactflow and sofar the experience is really good!

I'm trying to add react flow instances dynamically on a button click which worked, and I'm wrapping each my component of react flow with <ReactFlowProvider> and i'm passing a unique id to the reactflow component and appending the same id to the nodes and edges and creating unique node id and edge id.

As mentioned earlier the reactflow instance is rendering dynamically in tabs, whenever I switch tabs the build tree is reset to its initial state.

can anyone suggest me solution for this

**Comment by bcakmakoglu** [comment]
What exactly is the *issue* though? The reset to initial state?
Can you provide a reproduction for this? [(a sandbox template can be found here)](https://new.reactflow.dev/react-ts)
